### DISCUSSION TITLE
Access "fs" from boot file not working

### DISCUSSION THREAD

**Discussion by DarkLite1** [quasarframework/quasar#17596] [General - Components / Directives / etc] [answered]
We have a folder structure for languages in `src/i18n`:
- FRA\en.json
- FRA\fr.json
- DEU\de.json

The plan is to grab the folder names and use them in our store.

### Quasar boot file "src\boot\init.ts":
```typescript
import { store } from 'src/store'
import fs from 'fs'

const directories = (source: string) =>
  fs
    .readdirSync(source, {
      withFileTypes: true,
    })
    .filter((c) => c.isDirectory())
    .map((c) => c.name)

console.log('dir list', directories('src/i18n'))
```
This throws the following error:

> [Quasar]` boot error: Error: Module "fs" has been externalized for browser compatibility and cannot be accessed in client code.

In the [Quasar config file docs](https://quasar.dev/quasar-cli-vite/quasar-config-file#setting-env-for-dev-build) it states:

![image](https://github.com/user-attachments/assets/e1424a09-339d-4962-aa9b-7647a968ff32)

So I expected the boot file being able to access these folders or use the "fs" module in the context of Node. What is the correct way of retrieving the folder names and getting them in our store?

Thank you for your help.

**Comment by yusufkandemir** [comment, marked as answer]
quasar.config file is a configuration file, which runs at build time
Boot files are part of your application runtime code, so it runs in the browser

So, the docs do not say you can use it in boot files. You just misinterpreted it.

To achieve your goal, I see two options:
- since a list of available locales is simple data, you can calculate it in quasar.config and set an `env` variable. Then, parse/use that env variable in your store.
- use a Vite glob import: https://vite.dev/guide/features.html#glob-import
